language/expressions.md

```diff
@@ -1385,6 +1474,98 @@ Examples:
 </td></tr>
 </table>
 
+#### Timespan Type [ ]
+
+Produces a new `Timespan` type with a range. A `Timespan` type has the default range -Infinity to +Infinity.
+A `Timespan` range where one or both ends is Infinity is said to be an *open range*, else it is a
+*closed range*. The set of values in the range is inclusive of the given values.
+
+Signature:
+
+    TimespanTypeAccess
+      : Expression<Type<Timespan>> '[' exact = TimespanRangeValue ']'
+      | Expression<Type<Timespan>> '[' from = TimespanRangeValue ',' to = TimespanRangeValue ']'
+      ;
+      
+    TimespanRangeValue
+      : Expression<Timespan>
+      | Expression<Initializer[Timespan]>
+      | Expression<String>
+      | Expression<Integer>
+      | Expression<Float>
+      | 'default'
+      ;  
+
+* Accepts one or two keys
+* Keys must evaluate to a `Timespan`, an `Initializer[Timespan]`, a `String`, a `Float`, an `Integer`, or to literal `default`
+* A key that evaluates to `Initializer[Timespan]`, a `String`, a `Float`, or an `Integer`, will be coerced into a `Timespan` instance
+* A value of default means -Infinity if given as the value of `from`, and +Infinity if given
+  as `to`. If only an exact value of `default` is given this is the same as
+  `Timespan[default, default]`,
+  which again is the same as just `Timespan` (i.e. all floating point values in the range +/- Infinity).
+* The `from` value may be greater than the `to` value.
+* The range values may be negative.
+
+Examples:
+
+    Timespan[2]                        # a timespan of 2 seconds
+    Timespan[77.3]                     # a timespan of 1 minute, 17 seconds, and 300 milliseconds
+    Timespan[{hour => 1}, {hour => 2}] # a timespan between 1 and two hours
+    Timespan['1-00:00:00', '2-00:00:00'] # a timespan between 1 and 2 days
+    Timespan[Timespan('11', '%H'), Timespan('12', '%H')]              # a timespan between 11 and 12 hours
+    Timespan['1-00:00:00', Timespan({days => 2, nano-seconds => -1})] # a timespan between 1 and up to (but not including) 2 days
```

Comment:
How about just `nanoseconds`? I've always seen it treated as a single word, e.g. at https://en.wikipedia.org/wiki/Nanosecond.